model/DeliveryServerService.php

```diff
@@ -0,0 +1,304 @@
+        }
+        return $allowed;
+    }
+
+    /**
+     * Get delivery property values
+     * @param core_kernel_classes_Resource $delivery
+     * @return array
+     */
+    private function getDeliveryProperties(core_kernel_classes_Resource $delivery)
+    {
+        if (isset($this->deliveryProperties[$delivery->getUri()])) {
+            return $this->deliveryProperties[$delivery->getUri()];
+        }
+
+        $properties = [];
+
+        $repeatedDeliveryService =  $this->getServiceManager()->get(RepeatedDeliveryService::CONFIG_ID);
+
+        if ($repeatedDeliveryService->isRepeated($delivery)) {
+            $repeatedDelivery = $delivery;
+            $numberOfRepetitionProp = new core_kernel_classes_Property(RepeatedDeliveryService::PROPERTY_NUMBER_OF_REPETITION);
+            $numberOfRepetition = (string) $repeatedDelivery->getOnePropertyValue($numberOfRepetitionProp);
+            $delivery = $repeatedDeliveryService->getParentDelivery($delivery);
+        }
+
+        $deliveryProps = $delivery->getPropertiesValues(array(
+            new core_kernel_classes_Property(TAO_DELIVERY_MAXEXEC_PROP),
+            new core_kernel_classes_Property(TAO_DELIVERY_START_PROP),
+            new core_kernel_classes_Property(TAO_DELIVERY_END_PROP),
+            new core_kernel_classes_Property(DeliveryScheduleService::TAO_DELIVERY_RRULE_PROP),
+        ));
+
+        $propMaxExec = current($deliveryProps[TAO_DELIVERY_MAXEXEC_PROP]);
+        $propStartExec = current($deliveryProps[TAO_DELIVERY_START_PROP]);
+        $propEndExec = current($deliveryProps[TAO_DELIVERY_END_PROP]);
+        $rrule = current($deliveryProps[DeliveryScheduleService::TAO_DELIVERY_RRULE_PROP]);
+
+        if (isset($repeatedDelivery)) {
+            $rEvents = $repeatedDeliveryService->getRecurrenceCollection($delivery);
+            if (isset($rEvents[$numberOfRepetition])) {
+                $properties[TAO_DELIVERY_START_PROP] = $rEvents[$numberOfRepetition]->getStart()->getTimestamp();
+                $properties[TAO_DELIVERY_END_PROP] = $rEvents[$numberOfRepetition]->getEnd()->getTimestamp();
+                $properties[RepeatedDeliveryService::PROPERTY_NUMBER_OF_REPETITION] = $numberOfRepetition;
+            }
+        } else {
+            $properties[TAO_DELIVERY_START_PROP] = (!(is_object($propStartExec)) or ($propStartExec=="")) ? null : $propStartExec->literal;
+            $properties[TAO_DELIVERY_END_PROP] = (!(is_object($propEndExec)) or ($propEndExec=="")) ? null : $propEndExec->literal;
+            $properties[DeliveryScheduleService::TAO_DELIVERY_RRULE_PROP] = (!(is_object($rrule)) or ($rrule=="")) ? null : $rrule->literal;
+        }
+
+        $properties[TAO_DELIVERY_MAXEXEC_PROP] = (!(is_object($propMaxExec)) or ($propMaxExec=="")) ? 0 : $propMaxExec->literal;
+        $properties[CLASS_COMPILEDDELIVERY] = $delivery;
```

Comment:
I have pushed new commit, please review it again.
Please make sure that this PR merged before testing:
https://github.com/oat-sa/extension-tao-delivery/pull/131